snippets/machine-set-ifeval-open.adoc

```diff
@@ -0,0 +1,491 @@
+:api-version: v1beta1
+:api-kind: MachineSet
+:namespace: openshift-machine-api
+:providerspec-path: spec.template.spec.providerSpec.value
+endif::mapi[]
+
+// IBM Power VS
+
+ifeval::["{context}" == "mapi-compute-config-options-ibm-power-vs"]
+:mapi:
+endif::[]
+
+ifdef::mapi[]
+:provider-name: IBM Power VS
+:machine-set: compute machine set
+:api-name: Machine API
+:api-group: machine.openshift.io
+:api-version: v1beta1
+:api-kind: MachineSet
+:namespace: openshift-machine-api
+:providerspec-path: spec.template.spec.providerSpec.value
+endif::mapi[]
+
+// Nutanix
+
+ifeval::["{context}" == "mapi-compute-config-options-nutanix"]
+:mapi:
+endif::[]
+
+ifdef::mapi[]
+:provider-name: Nutanix
+:machine-set: compute machine set
+:api-name: Machine API
+:api-group: machine.openshift.io
+:api-version: v1beta1
+:api-kind: MachineSet
+:namespace: openshift-machine-api
+:providerspec-path: spec.template.spec.providerSpec.value
+endif::mapi[]
+
+// OpenStack
+
+ifeval::["{context}" == "mapi-compute-config-options-openstack"]
+:mapi:
+endif::[]
+
+ifdef::mapi[]
+:provider-name: RHOSP
+:machine-set: compute machine set
+:api-name: Machine API
+:api-group: machine.openshift.io
+:api-version: v1beta1
+:api-kind: MachineSet
+:namespace: openshift-machine-api
+:providerspec-path: spec.template.spec.providerSpec.value
+endif::mapi[]
+
+// VMware vSphere
+
+ifeval::["{context}" == "mapi-compute-config-options-vsphere"]
+:mapi:
+endif::[]
+
+ifdef::mapi[]
+:provider-name: vSphere
+:machine-set: compute machine set
+:api-name: Machine API
+:api-group: machine.openshift.io
+:api-version: v1beta1
+:api-kind: MachineSet
+:namespace: openshift-machine-api
+:providerspec-path: spec.template.spec.providerSpec.value
+endif::mapi[]
+
+// bare metal
+
+ifeval::["{context}" == "mapi-compute-config-options-bare-metal"]
+:mapi:
+endif::[]
+
+ifdef::mapi[]
+:provider-name: bare metal
+:machine-set: compute machine set
+:api-name: Machine API
+:api-group: machine.openshift.io
+:api-version: v1beta1
+:api-kind: MachineSet
+:namespace: openshift-machine-api
+:providerspec-path: spec.template.spec.providerSpec.value
+endif::mapi[]
+
+// #############################################################
+// # Cluster API compute machine set opening ifeval directives #
+// #############################################################
+
+// Agnostic
+// Adding a section to evaluate whether this would be helpful for the base specs.
+
+ifeval::["{context}" == "capi-compute-config-options-CLOUD"]
+:capi:
+endif::[]
+
+ifdef::capi[]
+:provider-name: <provider_name>
+:machine-set: compute machine set
+:api-name: Cluster API
+:api-group: cluster.x-k8s.io
+:api-version: v1beta1
+:api-kind: MachineSet
+:namespace: openshift-cluster-api
+:spec-path: spec.template.spec
+// CAPI machine template and machine set spec paths are identical
+endif::capi[]
+
+// Amazon Web Services (AWS)
+
+ifeval::["{context}" == "capi-compute-config-options-aws"]
+:capi:
+endif::[]
+
+ifdef::capi[]
```

Comment:
I noticed starting here that the entries include `template-` fields. I assume that's intentional, but I just wanted to note it in case it's not.

Reply:
Thanks for checking! This comment demonstrates I missed the template section on the immediately previous section - great eye!

Reply:
Glad I could help!